service/src/main/java/tds/config/repositories/impl/ConfigRepositoryImpl.java

```diff
@@ -113,11 +135,60 @@ public ClientTestProperty getClientTestProperty(final String clientName, final S
                 "WHERE\n" +
                 "   clientname = :clientName\n" +
                 "   AND testid = :assessmentId";
+
+        Optional<ClientTestProperty> clientTestPropertyOptional;
         try {
-            return jdbcTemplate.queryForObject(SQL, parameters, new BeanPropertyRowMapper<>(ClientTestProperty.class));
+            clientTestPropertyOptional = Optional.of(
+                    jdbcTemplate.queryForObject(
+                            SQL,
+                            parameters,
+                            new BeanPropertyRowMapper<>(ClientTestProperty.class)));
         } catch (EmptyResultDataAccessException e) {
             LOG.debug("{} did not return results for clientName = {}, assessmentId = {}", SQL, clientName, assessmentId);
-            return null;
+            clientTestPropertyOptional = Optional.empty();
+        }
+
+        return clientTestPropertyOptional;
+    }
+
+    @Override
+    public Optional<TimeLimits> getTimeLimits(final String clientName) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("clientName", clientName);
+
+        final String SQL = TIMELIMITS_BASE_SQL + "   AND _efk_testid IS NULL";
```

Comment:
It was worth a shot, but I agree w/you and Todd - it makes the code harder to read/understand.  I've backed this out and put the queries back in their respective methods.